pkg/controller/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -0,0 +1,251 @@
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to secondary resource TemplateUpdateRequest and requeue the owner NSTemplateTier
+	err = c.Watch(&source.Kind{Type: &toolchainv1alpha1.TemplateUpdateRequest{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &toolchainv1alpha1.NSTemplateTier{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileNSTemplateTier implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileNSTemplateTier{}
+
+// ReconcileNSTemplateTier reconciles a NSTemplateTier object
+type ReconcileNSTemplateTier struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client                 client.Client
+	scheme                 *runtime.Scheme
+	retrieveMemberClusters cluster.GetMemberClustersFunc
+}
+
+// Reconcile reads that state of the cluster for a NSTemplateTier object and makes changes based on the state read
+// and what is in the NSTemplateTier.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileNSTemplateTier) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	logger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	logger.Info("Reconciling NSTemplateTier")
+
+	// Fetch the NSTemplateTier instance
+	instance := toolchainv1alpha1.NSTemplateTier{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, &instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		logger.Error(err, "unable to get the current NSTemplateTier instance")
+		return reconcile.Result{}, err
+	}
+
+	activeTemplateUpdateRequests, err := r.activeTemplateUpdateRequests(instance)
+	if err != nil {
+		logger.Error(err, "unable to get active TemplateUpdateRequests")
+		return reconcile.Result{}, err
+	}
+
+	if activeTemplateUpdateRequests < MaxPoolSize {
+		members := r.retrieveMemberClusters()
+	loop:
+		for _, member := range members {
+			// create a TemplateUpdateRequest if active count < max pol size
+			// find a MasterUserRecord which is not already up-to-date
+			// and for which there is no TemplateUpdateRequest yet
+			// fetch by subsets of "MaxPoolSize + 1" size until a MasterUserRecord candidate is found
+			murs := toolchainv1alpha1.MasterUserRecordList{}
+			matchingLabels, err := newLabelSelector(instance, *member)
+			if err != nil {
+				logger.Error(err, "unable to get MasterUserRecords to update")
+				return reconcile.Result{}, err
+			}
+			r.client.List(context.Background(), &murs,
```

Comment:
golangci-lint would caught it... you missed the error handling.

Reply:
fixed in 776b35b